src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -57,13 +57,30 @@ public static Process Start(string fileName, string arguments, string userName,
         /// <summary>Terminates the associated process immediately.</summary>
         public void Kill()
         {
-            EnsureState(State.HaveNonExitedId);
-            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGKILL) != 0)
+            EnsureState(State.HaveId);
+
+            // Don't do anything if we know the process has exited.
+            // This also protects us from killing another process with a recycled PID.
+            if (GetHasExited(refresh: false))
+            {
+                return;
+            }
+
+            int killResult = Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGKILL);
+            if (killResult != 0)
             {
-                throw new Win32Exception(); // same exception as on Windows
+                Interop.Error error = Interop.Sys.GetLastError();
+                // Don't throw if the process has exited.
+                if (error != Interop.Error.ESRCH)
+                {
+                    throw new Win32Exception();
```

Comment:
You didn't want to keep the comment?

Reply:
Not an active decision. I'll add it back.